xonsh/history.py

```diff
@@ -70,6 +70,15 @@ def _gc_bytes_to_rmfiles(hsize, files):
     return rmfiles
 
 
+def _get_history_files(reverse=False):
+    data_dir = builtins.__xonsh_env__.get('XONSH_DATA_DIR')
+    data_dir = expanduser_abs_path(data_dir)
+    files = [os.path.join(data_dir, f) for f in os.listdir(data_dir)
+             if f.startswith('xonsh-') and f.endswith('.json')]
+    files.sort(key=lambda x: os.path.getmtime(x), reverse=reverse)
```

Comment:
Maybe, but you probably also need to filter out the locked files, which means you need to open up the file anyway. Or does that filtering happen later?